unity-ggjj/Assets/Scripts/TextDecoder/DirectorActionDecoder.cs

```
@@ -29,7 +28,7 @@ public void OnNewActionLine(string line)
         catch (TextDecoder.Parser.ScriptParsingException exception)
         {
             Debug.LogError(exception);
-            _onActionDone.Invoke();
+            _narrativeScriptPlayer.Continue();
```

I believe this entire `catch` block (and wrapping the code in question in a `try/catch`-block in general) is actually no longer necessary. If we encounter a line that fails to parse, I would expect the game to stop.

Right now we would silently continue the game, which I think is a bit dangerous, given we've failed to parse an action line. (According to Git I actually added this but I'm not entirely sure why we needed it in the first place. 🤔)

---

If we remove it, the game will essentially softlock. Maybe we should program in some kind of error screen if the catch block is reached. I think this should be a separate issue though. It's been kind of useful to have this feature tbh as it means i can put actions in narrative scripts which haven't been added yet, though that's admittedly less relevant now.

---

So I just tested it this out and uncaught exceptions actually... don't crash the game. That's uhhh... news to me. 👀💦  
**TIL:** As Unity apps are compile to native code and not [CLI apps](https://en.wikipedia.org/wiki/Common_Language_Infrastructure), uncaught exceptions just... vanish.

This was actually my main reason for using exceptions over logs for any state that might happen during development but is unexpected during regular gameplay: I expected the game to simply end at that point. And that's actually not the case with Unity by default. Which I kind I just assumed; that's my bad!  
(Web Player builds also couldn't even "be killed" similarly to how `Application.Quit()` doesn't work there, so we would have needed at least some form of custom error handling for Web Player builds anyway, so I like your idea of adding some form of error handling.)  
Given that: Yes, we **absolutely** need some form of error handling.

I'll create a discussion for it, so we can discuss how we want this to look both from a code and UX perspective.  
(That all said: In my opinion there is a [fairly easy] way to do it, I also already have a rough idea for it, I just need to sketch it out and put up the thread for it.)

---

Keeping that in mind, I would still propose following the mental model that we treat uncaught exceptions as "fatal errors we can't recover from", even if we need to add a bit of custom code for that to apply. The advantage being, that this allows us to cover not only "happy paths" for our logic (by deliberately placing exceptions), making errors a bit more human readable (like the `ScriptParsingException` for example) and also allows us to handle them gracefully in our code (with things like `Assert.Throws(() => { ... })`).

The way I see it, if an action doesn't exist in the decoder (even if it might get added later on), it simply isn't implemented (yet), thereby also shouldn't be used in narrative scripts (yet) and the game cannot assume it can continue in a way we would expect it to, if it encounters a line it can't parse. (Imagine a piece evidence not being added, but the story still progressing to a cross examination where it would that piece is required to progress with the trial.)  
Plus, with https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/191 already integrated, it will even fail builds and throw errors in the editor, if a script with not-supported actions is present; I'd like to avoid treating the severity of specific errors differently during compile-time and run-time.

(I really hope this doesn't come across as pedantic. One of the last apps I worked on had no organized way of handling errors which made bugs nearly untraceable, so it's a subject I'm pretty passionate about. I've seen it cause a lot of headache and would avoid it if possible. 👀)

---

Split this up into two discussions actually: [One about the UX](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/discussions/198), the other about [the technical side of things](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/discussions/199).